src/App.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import './App.css';
+import Header from './components/Header';
+import ItemList from './components/ItemList';
+import Selector from './components/Selector';
+import mockData from './mockData/restaurantList.json';
+import { appState, restaurant } from './utils/interfaces';
+import { parseJson } from './utils/json';
+import { selectorCategory, selectorFilter } from './utils/types';
+import { sortingByCategory, sortingByFilter } from './domain/restaurantSort';
+import { CATEGORY_OPTIONS, FILTER_OPTIONS } from './utils/constants';
+import { localStorageGetItem } from './utils/localStorage';
+import { typePredicates } from './utils/typeCheck';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface Props {}
+
+class App extends React.Component<Props, appState> {
+  constructor(props: Props | Readonly<Props>) {
+    super(props);
+
+    const localStorageSavedList = typePredicates<Array<restaurant>>({
+      data: parseJson(JSON.stringify(localStorageGetItem('restaurantList'))),
+      initialData: [],
+    });
+
+    const restaurantMockDataList = typePredicates<Array<restaurant>>({
```

Comment:
소소하지만 요렇게도 사용 가능하답니다! 알고 계실 수도 있지만 혹시나 모르실까봐 코멘트 추가합니다!
```suggestion
    const restaurantMockDataList = typePredicates<restaurant[]>({
```